core/commands/diff.py

```diff
@@ -52,14 +55,36 @@ class GsDiffRefreshCommand(TextCommand, GitCommand):
     """
 
     def run(self, edit, cursors=None):
-        in_cached_mode = self.view.settings().get("git_savvy.diff_view.in_cached_mode")
-        base_commit = self.view.settings().get("git_savvy.diff_view.base_commit")
+        in_cached_mode      = self.view.settings().get("git_savvy.diff_view.in_cached_mode")
+        in_ignore_all_space = self.view.settings().get("git_savvy.diff_view.in_ignore_all_space")
+        in_word_diff        = self.view.settings().get("git_savvy.diff_view.in_word_diff")
+        base_commit         = self.view.settings().get("git_savvy.diff_view.base_commit")
 
         stdout = self.git(
-            "diff", "--no-color", base_commit,  "--cached" if in_cached_mode else None, "--", self.file_path)
+            "diff",
+            "--ignore-all-space" if in_ignore_all_space else None,
+            "--word-diff" if in_word_diff else None,
+            "--no-color", base_commit,
+            "--cached" if in_cached_mode else None,
+            "--",
+            self.file_path)
 
         self.view.run_command("gs_replace_view_text", {"text": stdout})
 
+class GsDiffSetSettingAndRefresh(WindowCommand):
+
+    """
+    Set a setting to the diff view like: in_ignore_all_space, in_word_diff
+    Refresh the view after
+    """
+
+    def run(self, setting):
+        setting_str = "git_savvy.diff_view.{}".format(setting)
+        print(setting_str)
+        view = self.window.active_view()
```

Comment:
By changing this to `TextCommand` subclass, no need to check `self.window` for the active view.  Just use `self.view`.